apps/meteor/app/reactions/server/setReaction.js

```diff
@@ -3,16 +3,40 @@ import { TAPi18n } from 'meteor/rocketchat:tap-i18n';
 import _ from 'underscore';
 import { EmojiCustom } from '@rocket.chat/models';
 
-import { Messages, Rooms } from '../../models/server';
+import { Messages, Rooms, Users } from '../../models/server';
 import { callbacks } from '../../../lib/callbacks';
 import { emoji } from '../../emoji/server';
 import { isTheLastMessage, msgStream } from '../../lib/server';
 import { canAccessRoom, hasPermission } from '../../authorization/server';
 import { api } from '../../../server/sdk/api';
 import { AppEvents, Apps } from '../../apps/server/orchestrator';
+import { settings } from '../../settings/server';
+
+const getNameByUsername = (username, message, reaction) => {
+	const names = new Map();
+	Users.findUsersByUsernames(message.reactions[reaction].usernames, {
+		fields: {
+			username: 1,
+			name: 1,
+		},
+	}).forEach((data) => {
+		names.set(data.username, data.name);
+	});
+	return names.get(username) || username;
+};
```

Comment:
We only have the user object for the `user` who is reacting, right? The idea was to add names for users who have already reacted as well, using their usernames.

Reply:
That should be a migration, not part of the running code. It's a one-off task. Or even we could just leave it empty and let the frontend fall back to username for old reactions. Why? Because the migration can be quite costly.